### PR TITLE
Host a local repo of our yum dependencies

### LINKED ISSUE #5161: Deploy local yum repo
User statement:  As an engineer on VIC, I would like a build process that is fast and consistent. Local repos will help with both of those goals.

Details:
https://www.digitalocean.com/community/tutorials/how-to-set-up-and-use-yum-repositories-on-a-centos-6-vps#custom-yum-repository

Acceptance Criteria:
1. Builds are done using local repos.

### PR DESCRIPTION
**Pull request by mhagen-vmware**
I have skipped CI because I am still struggling to host this within our CI on a VCH, but I have tested it locally and it works great.  Build times are down from about 13 minutes to 6.5 minutes.  Not going to merge this until I can get the CI VCH up and running and can test this within CI, but this is ready for review.

fixes #5161

### CODE REVIEW

File: isos/base/photon-updates-local.repo [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[photon-updates]
+name=VMware Photon Linux 1.0(x86_64)
+baseurl=ftp://192.168.31.16/photon-updates
```
Comment (caglar10ur) [inline isos/base/photon-updates-local.repo]: where is that IP coming from? Could it be a different for different runs?

Comment (mhagen-vmware) [inline isos/base/photon-updates-local.repo]: This is the static IP address that I have reserved in the CI, it won't change.

File: infra/integration-image/LocalRepo [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+RUN mkdir /photon && mkdir /photon-updates
+
+ENV EXCLUDE_LIST "index.html*,openjdk*,openjre*,perl*,postgresql*,python-*,\
+python3*,ruby*,subversion*,gnome*,NetworkManager*,cloud*,docker*,grub*,ktap*,\
+kubernetes*,linux-docs*,linux-sound*,linux-tools*,docbook*,httpd*,go-*,jna*,\
+linux-debuginfo*,linux-dev*,linux-docs*,linux-drivers*,linux-oprofile*,linux-sound*,\
+linux-tools*,linux-esx-debuginfo*,linux-esx-devel*,linux-esx-docs*,nginx*,sysdig*"
+
+RUN wget -e robots=off -r -nH -nd -np -R $EXCLUDE_LIST https://vmware.bintray.com/photon_release_1.0_x86_64/x86_64/ -P /photon
+RUN wget -e robots=off -r -nH -nd -np -R $EXCLUDE_LIST https://dl.bintray.com/vmware/photon_updates_1.0_x86_64/x86_64/ -P /photon-updates
```
Comment (mhagen-vmware) [inline infra/integration-image/LocalRepo]: Yea, I had that and would prefer that as well, unfortunately that ends up being over 12GB image. Which is incredibly unwieldy (both upload/download and filling up our harbor instance).  It also takes almost 2 hrs to build the image if we don't exclude.

Comment (mhagen-vmware) [inline infra/integration-image/LocalRepo]: And I am a bit concerned about hammering their repo with a crawler that pulls from them for 2hrs straight.

Comment (caglar10ur) [inline infra/integration-image/LocalRepo]: That 12GB is a one time cost though right? After that we can use rsync or something like that to get only changed files

**Comment by mhagen-vmware** [comment]
Me got angry with FTP and now it is HTTP :)

**Comment by frapposelli** [comment]
@mhagen-vmware why not hosting the rpms in a host directory/docker volume and having a cron that updates it? so you can feed the rpm list independently of the container.

**Comment by mhagen-vmware** [comment]
@frapposelli Yea, I was considering that. It would be a nice addition, but according to the original issue #5161 we actually want to pin the version per release.  And that I think would put even more stress on the original repo as well if multiple people start using this and hammering the repo with regularity.  I think the biggest goal though is to not update our dep at random times though, so that we don't get a build break because they change something major in photon in the middle or even worse on the last day of our release cycle.

**Pull request merged**